include/oneapi/dpl/pstl/iterator_impl.h

```diff
@@ -424,7 +424,9 @@ class transform_iterator
     typedef typename ::std::iterator_traits<_Iter>::pointer pointer;
     typedef typename ::std::iterator_traits<_Iter>::iterator_category iterator_category;
 
-    transform_iterator(_Iter __it = _Iter(), _UnaryFunc __unary_func = _UnaryFunc())
+    transform_iterator() = default;
```

Comment:
Is this new constructor really needed?
If yes, should it be explicit?

Reply:
Yes, it is needed.
As I mentioned in the description:

"We are also required to make change in transform_iterator default constructor, because calling the default constructor of the std::optional with the type which default constructor is ill-formed (non-deleted) is also ill-formed on clang (see this example: https://godbolt.org/z/MM93c5GWx). Because of this we need the default constructor to be deleted if the body of the iterator is not default constructible. From the first glance, it is still aligned with the oneDPL specification."

I will try to formulate reasonable comment for the code before merging this.

Reply:
what about `explicit` ?

Reply:
This part was removed from the PR.
In any case, `explicit` is orthogonal and would not fix the previous issue.

Reply:
ok